test/AutoRest.TestServer.Tests/ProtocolMethodsInRestClientTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Linq;
+using Azure;
+using NUnit.Framework;
+using ProtocolMethodsInRestClient;
+
+namespace AutoRest.TestServer.Tests
+{
+    internal class ProtocolMethodsInRestClientTests
+    {
+        [TestCase("Delete", true)]
+        [TestCase("Create", true)]
+        [TestCase("Get", false)]
+        public void ProtocolMethodGeneratedInRestClient(string methodName, bool isProtocolMethodGenerated)
+        {
+            var methods = typeof(TestServiceRestClient).GetMethods();
+            Assert.IsNotNull(methods);
+
+            var restClientMethods = methods.Where(m => m.Name.Equals(methodName));
+            if (isProtocolMethodGenerated)
```

Comment:
It feels to me like we should be asserting that isProtocolMethodGenerated is equal to a value we learn from the generated client, rather than conditionally asserting based on its value.

Reply:
[Here](https://github.com/Azure/autorest.csharp/blob/7d3f3ac43b97f4921f5b3f9c38202f11275ea60d/test/AutoRest.TestServer.Tests/ProtocolMethodsInRestClientTests.cs#L13-L15) I added the method names and an expected flags to verify that the protocol method should generate for this or not. [Here](https://github.com/Azure/autorest.csharp/blob/7d3f3ac43b97f4921f5b3f9c38202f11275ea60d/test/AutoRest.TestServer.Tests/ProtocolMethodsInRestClientTests.cs#L35) I'm asserting in [two of the methods](https://github.com/Azure/autorest.csharp/blob/7d3f3ac43b97f4921f5b3f9c38202f11275ea60d/test/AutoRest.TestServer.Tests/ProtocolMethodsInRestClientTests.cs#L24), one is generated as protocol method.

Reply:
I guess my concern is that if the wrong value was passed in for isProtocolMethodGenerated, we would check the wrong test case.  It also seems strange [here](https://github.com/Azure/autorest.csharp/blob/7d3f3ac43b97f4921f5b3f9c38202f11275ea60d/test/AutoRest.TestServer.Tests/ProtocolMethodsInRestClientTests.cs#L40), that we're effectively saying

```
if (isProtocolMethodGenerated == false)
{
    Assert.IsFalse(isProtocolMethodGenerated);
}
```

Reply:
`isProtocolMethodGenerated` values refers to the method names added in the config and that's how I'm verifying that these methods are generated as protocol methods. Otherwise, how would we know which methods we wants to verify?

I've removed:
` Assert.IsFalse(isProtocolMethodGenerated);`

Reply:
`isProtocolMethodGenerated` values refer to the method names added in the config(true, if method name added in the config, false, if not) to verify that these methods are generated as protocol methods. Otherwise how do we verify these methods?

I've removed:
`Assert.IsFalse(isProtocolMethodGenerated);`

Reply:
Would it make sense to split this into two separate test cases, rather than branching on `isProtocolMethodGenerated`?

Reply:
Updated!